npm/src/language-service/language-service.ts

```diff
@@ -165,33 +196,46 @@ export class QSharpLanguageService implements ILanguageService {
     documentUri: string,
     offset: number,
   ): Promise<IDefinition | undefined> {
-    let code = this.code[documentUri];
-    if (code === undefined) {
+    const sourceCode = this.code[documentUri];
+    if (sourceCode === undefined) {
       log.error(
         `getDefinition: expected ${documentUri} to be in the document map`,
       );
       return undefined;
     }
-    const convertedOffset = mapUtf16UnitsToUtf8Units([offset], code)[offset];
+    const convertedOffset = mapUtf16UnitsToUtf8Units([offset], sourceCode)[
+      offset
+    ];
     const result = this.languageService.get_definition(
       documentUri,
       convertedOffset,
     );
     if (result) {
-      // Inspect the URL protocol (equivalent to the URI scheme + ":").
-      // If the scheme is our library scheme, we need to call the wasm to
-      // provide the library file's contents to do the utf8->utf16 mapping.
-      const url = new URL(result.source);
-      if (url.protocol === qsharpLibraryUriScheme + ":") {
-        code = wasm.get_library_source_content(url.pathname);
-        if (code === undefined) {
-          log.error(`getDefinition: expected ${url} to be in the library`);
-          return undefined;
+      let targetCode = this.code[result.source];
```

Comment:
Do we need to be concerned about the edge case where there is a user source with the same name as a library source? Would they have the same URI except the library source would have the scheme prepending it?

Reply:
They would both be full URIs, with different schemes, so I believe we're safe.

Example:
`vscode-test-web://mount/intrinsic.qs`
`qsharp-library-source:intrinsic.qs`